base/db/R/convert_input.R

```diff
@@ -512,66 +510,17 @@ convert_input <-
         # we'll need to update its start/end dates .
       } 
     } else {
-      # No existing record found. Should be good to go.
+      PEcAn.logger::logger.debug("No existing record found. Should be good to go.")
     }
   }
   
   #---------------------------------------------------------------------------------------------------------------#
   # Get machine information
+  machine.info <- get.machine.info(host, input.args = input.args, input.id = input.id)
 
-  machine.host <- ifelse(host$name == "localhost", PEcAn.remote::fqdn(), host$name)
-  machine <- db.query(paste0("SELECT * from machines where hostname = '",
-                             machine.host, "'"), con)
-  
-  if (nrow(machine) == 0) {
-    PEcAn.logger::logger.error("machine not found", host$name)
-    return(NULL)
-  }
-  
-  if (missing(input.id) || is.na(input.id) || is.null(input.id)) {
-    input <- dbfile <- NULL
-  } else {
-    input <- db.query(paste("SELECT * from inputs where id =", input.id), con)
-    if (nrow(input) == 0) {
-      PEcAn.logger::logger.error("input not found", input.id)
-      return(NULL)
-    }
-    
-    if(!is.null(input.args$dbfile.id)){
-      dbfile <-
-        db.query(
-          paste(
-            "SELECT * from dbfiles where id=",input.args$dbfile.id," and container_id =",
-            input.id,
-            " and container_type = 'Input' and machine_id =",
-            machine$id
-          ),
-          con
-        )  
-    }else{
-      dbfile <-
-        db.query(
-          paste(
-            "SELECT * from dbfiles where container_id =",
-            input.id,
-            " and container_type = 'Input' and machine_id =",
-            machine$id
-          ),
-          con
-        )  
-    }
-    
-
-    
-    if (nrow(dbfile) == 0) {
-      PEcAn.logger::logger.error("dbfile not found", input.id)
-      return(NULL)
-    }
-    if (nrow(dbfile) > 1) {
-      PEcAn.logger::logger.warn("multiple dbfile records, using last", dbfile)
-      dbfile <- dbfile[nrow(dbfile), ]
-    }
-  }
+  machine <- machine.info$machine
```

Comment:
Aha, here's where my comment above about return values comes back:
* in the existing code, if we don't find any results when looking up the machine, input, or dbfile then the whole `convert.input` function will return NULL before performing any conversions
* with this change, if we don't find any results then these `machine`, `input`, and `dbfile` variables will all be NULL, but execution of `convert.input` will continue into the next section if not stopped by a `logger.error()`.

Might be worth checking for this just in case:
```suggestion
  if (any(sapply(machine.info, is.null))) {
    PEcAn.logger::logger.error("failed lookup of inputs or dbfiles")
    return(NULL)  
  }
  machine <- machine.info$machine
```